File: problems/4/solution.py
class Solution:
    def removeDuplicates(self, head):
        if not head or not head.next:
            return head

        current = head
        nextNode = current.next

        while nextNode and nextNode.val == current.val:
            nextNode = nextNode.next;

        current.next = self.removeDuplicates(nextNode)
        return current

File: problems/4/solution.java
// ListNode type
// class ListNode {
//     int val;
//     ListNode next;
//     ListNode(int x) { val = x; next = null; }
// }

class Solution {
    public static ListNode removeDuplicates(ListNode head) {
        if (head == null || head.next == null) {
            return head;
        }
        ListNode current = head;
        ListNode next = current.next;

	while (next != null && next.val == current.val) {
		next = next.next;
	}

	current.next = removeDuplicates(next);
	return current;
    }
}
